packages/platform/src/lib/signal-input.directive.ts

```diff
@@ -12,6 +12,16 @@ import {SIGNAL_INPUT_MODIFIER, SignalInputModifier} from "./signal-input-modifie
     '[class.control-valid]': 'this.formField?.state() === "VALID"',
     '[class.control-invalid]': 'this.formField?.state() === "INVALID"',
     '[class.control-pending]': 'this.formField?.state() === "PENDING"',
+    '[class.control-pristine]': 'this.formField?.dirtyState() === "PRISTINE"',
```

Comment:
I suppose we can remove the custom `class.control-state` classes with this change.

Reply:
Agreed. I will remove those and update the code accordingly